zmon-controller-app/src/main/java/org/zalando/zmon/api/EntityApi.java

```diff
@@ -67,13 +69,15 @@ public EntityApi(EntitySProcService entitySprocs, ObjectMapper mapper, MetricReg
     @ResponseBody
     @RequestMapping(value = {"/", ""}, method = {RequestMethod.PUT, RequestMethod.POST})
     public ResponseEntity<String> addEntity(@RequestBody JsonNode entity) {
-
-        if (entity.has("type")) {
-            if ("global".equals(entity.get("type").textValue().toLowerCase())) {
+        Optional<String> maybeNewType = Optional.ofNullable(entity.get("type")).map(JsonNode::textValue);
```

Comment:
is type and id really optional? I thought every entity must have both. @vetinari ?

Reply:
IIRC the current tests do not enforce this, some tests are just operating on an empty map (`{}`) which looks/feels wrong. The [database](https://github.com/zalando-zmon/zmon-controller/blob/master/database/zmon/10_data/04_tables/10_entity.sql#L7-L8) however enforces `id` and `type` to be non null and match a specific pattern... so we can safely set them as non optional.

Reply:
As discussed during standup, let's merge it like this and make changes to enforce `type` and `id` to be present in another PR.